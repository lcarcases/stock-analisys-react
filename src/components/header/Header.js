import React, { Component } from 'react';
import Search from './../search/Search';
import avatar from './../../assets/images/empty-avatar.jpg';
import styles from './Header.module.scss';

class Header extends Component {

    constructor(props) {
        super(props);

        this.state = {
            stockName: this.props.stockName,
            userName: this.props.userName,
            stockNameMarginLeft: ( this.props.headerStyles !== undefined &&
                                   this.props.headerStyles.stockName.marginLeft)
                                 ? this.props.headerStyles.stockName.marginLeft
                                 : '3rem',
            searchInputMarginLeft: ( this.props.headerStyles !== undefined &&
                                     this.props.headerStyles.searchInput.marginLeft
                                   )
                                   ? this.props.headerStyles.searchInput.marginLeft
                                   : '5rem',
            /*headerWidth: ( this.props.headerStyles !== undefined &&
                           this.props.headerStyles.header !== undefined &&
                           this.props.headerStyles.header.width
                         )
                         ? this.props.headerStyles.header.width
                         : '5rem', */
        }
    }

    render() {

        return (
                 <div className={styles.header}>
                    <div {...{ className:styles.stockName}} style={{marginLeft: this.state.stockNameMarginLeft}}>
                      <h2><span >{this.props.stockName}</span></h2>
                    </div>
                    <div {...{className:styles.searchComponent}} style={{marginLeft: this.state.searchInputMarginLeft}}>
                       <Search />
                    </div>
                    <div className={styles.user}>
                      <h2><span className={styles.user__name}>{this.props.userName ? this.props.userName: "" }</span></h2>
                      <img className={styles.user__avatar} src={avatar} alt='avatar'/>
                    </div>
                 </div>
        );
    }
}

export default Header;