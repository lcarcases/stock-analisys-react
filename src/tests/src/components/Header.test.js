import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import Header from '../../../components/header/Header';


describe('Pruebas Header', () => {
    
    test('debe de mostrar <Header /> correctamente', () => {

        

        const test = true;
        const username = 'lcarcases4';
        const stockName = "STOCKS ANALYSIS APP  ";

        let headerStyles = {};
        headerStyles.stockName = {};
        headerStyles.searchInput = {};

        headerStyles.stockName.marginLeft = '5rem';
        headerStyles.searchInput.marginLeft = '-1.8rem';
        if(window.innerWidth > 375 && window.innerWidth <= 900) {
            headerStyles.searchInput.marginLeft = '-7.8rem';
        }

        const wrapper = shallow(<Header className="header"
                                        stockName={stockName}
                                        userName={username}
                                        headerStyles={headerStyles}>
                                </Header>
                               );

        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        
        const p = "1234";

        const test = true;
        const username = 'lcarcases4';
        const stockName = "STOCKS ANALYSIS APP  ";

        let headerStyles = {};
        headerStyles.stockName = {};
        headerStyles.searchInput = {};

        headerStyles.stockName.marginLeft = '5rem';
        headerStyles.searchInput.marginLeft = '-1.8rem';
        if(window.innerWidth > 375 && window.innerWidth <= 900) {
            headerStyles.searchInput.marginLeft = '-7.8rem';
        }

        const wrapper = shallow(<Header className="header"
                                        stockName={stockName}
                                        userName={username}
                                        headerStyles={headerStyles}>
                                </Header>
                               );

        const spanText = wrapper.find('span').first().text();
        
        expect( spanText ).toBe( stockName );

    })
    
    
    

})