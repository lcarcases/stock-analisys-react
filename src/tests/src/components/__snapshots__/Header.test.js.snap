// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Header debe de mostrar <Header /> correctamente 1`] = `
<div
  className="header"
>
  <div
    className="stockName"
    style={
      Object {
        "marginLeft": "5rem",
      }
    }
  >
    <h2>
      <span>
        STOCKS ANALYSIS APP  
      </span>
    </h2>
  </div>
  <div
    className="searchComponent"
    style={
      Object {
        "marginLeft": "-1.8rem",
      }
    }
  >
    <Search />
  </div>
  <div
    className="user"
  >
    <h2>
      <span
        className="user__name"
      >
        lcarcases4
      </span>
    </h2>
    <img
      alt="avatar"
      className="user__avatar"
      src="empty-avatar.jpg"
    />
  </div>
</div>
`;
